unlock-app/src/components/interface/locks/CheckoutUrl/index.tsx

```diff
@@ -31,11 +33,46 @@ export const CheckoutUrlPage = () => {
     skipRecipient: true,
   })
 
-  const onAddLocks = (locks: any) => {
+  const handleRecurring = (locks: Record<string, any>) => {
+    let newLocksMapping: Record<string, any> = {
+      ...locks,
+    }
+    Object.entries(locks).map(async ([lockAddress, values]) => {
+      const { isRecurringPossible = false } = await getIsRecurringPossible({
+        lockAddress,
+        network: values!.network!,
+      })
+
+      // checkout config URL default to 1 year of recurring payments if recurring is set
+      if (isRecurringPossible) {
+        const recurringPayments =
+          values?.recurringPayments || (isRecurringPossible ? 12 : undefined)
```

Comment:
You can't use 12 here because this assumes the lock's duration is 1 month which may not be right! Rather you  should get the lock duration and then fund what number of renewals is required.

Reply:
Oh good point! fixed 

**Example 1**
<img width="450" alt="Screenshot 2023-03-20 at 12 04 27" src="https://user-images.githubusercontent.com/20865711/226341032-e6ea12fa-6696-4e0d-8863-0b375b0fc3f3.png">

**Example 2**
<img width="450" alt="Screenshot 2023-03-20 at 12 37 30" src="https://user-images.githubusercontent.com/20865711/226341041-11384b5d-9176-430f-b06c-c9ec07bbb13c.png">